insonmnia/hub/slot.go

```diff
@@ -0,0 +1,167 @@
+package hub
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	log "github.com/noxiouz/zapctx/ctxlog"
+	"github.com/pborman/uuid"
+	"github.com/sonm-io/core/insonmnia/structs"
+	pb "github.com/sonm-io/core/proto"
+	"github.com/sonm-io/core/util"
+	"go.uber.org/zap"
+)
+
+type AskPlan struct {
+	Id    string
+	Order *structs.Order
+}
+
+type AskPlansData map[string]*AskPlan
```

Comment:
Even though Go looks a bit like C, its compiler works differently. And C analogy does not always work with Go. Passing by value in Go may be significantly cheaper than passing by pointer. This happens because Go uses escape analysis to determine if variable can be safely allocated on function's stack frame, which could be much cheaper then allocating variable on the heap. Passing by value simplifies escape analysis in Go and gives variable a better chance to be allocated on the stack.

In other words, it's not really necessary to use a pointer to AskPlan in this case.
I've made an example (even though it's quite primitive, it shows the case) 
https://paste.ofcode.org/tD3nMDxewkYk2ZrVc9eipY

```
❯ go test -bench=.
goos: darwin
goarch: amd64
pkg: github.com/screwyprof/refvalue
BenchmarkByRef-8   	300000000	         5.47 ns/op
BenchmarkByVal-8   	300000000	         5.37 ns/op
PASS
ok  	github.com/screwyprof/refvalue	4.355s
```

Reply:
I feel it was here to allow nullable state, it's go, remember? No `Option<AskPlan>` , no `AskPlan?`... and this is sad.
However exactly in this case it doesn't matter, except implicit sharing (which doesn't seems).

Vote for removing pointer.